schedule/models/events.py

```diff
@@ -136,7 +225,7 @@ def _get_occurrence_list(self, start, end):
             if self.end_recurring_period and self.end_recurring_period < end:
                 end = self.end_recurring_period
 
-            rule = self.get_rrule_object()
+            rule = self.get_optimized_rrule_object(start-difference)
```

Comment:
Can you make this conditional based on a setting so that we can easily switch it off if necessary? I.e. `OPTIMIZED_RRULE_GENERATION = get_boolean('OPTIMIZED_RRULE_GENERATION')` in `settings.py`